crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/util/GenerateCsvUtil.java

```diff
@@ -0,0 +1,59 @@
+package uk.gov.justice.laa.crime.crowncourt.util;
+
+import lombok.experimental.UtilityClass;
+import lombok.extern.slf4j.Slf4j;
+import uk.gov.justice.laa.crime.crowncourt.entity.ReactivatedProsecutionCase;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.attribute.FileAttribute;
+import java.nio.file.attribute.PosixFilePermission;
+import java.nio.file.attribute.PosixFilePermissions;
+import java.util.List;
+import java.util.Set;
+
+@UtilityClass
+@Slf4j
+public class GenerateCsvUtil {
+
+    private static final String FILE_HEADER = "maat id, case URN, hearing id, previous outcome, previous outcome date, date of status change" + System.lineSeparator();
+    private static final String FILE_PERMISSIONS = "rwx------";
+
+    public File generateCsvFile(List<ReactivatedProsecutionCase> reactivatedCaseList, String fileName) throws IOException {
+        File targetFile = createCsvFile(fileName);
+        try (FileWriter fw = new FileWriter(targetFile, true)) {
+            // Write the CSV header
+            fw.append(FILE_HEADER);
+            // Write each ReactivatedProsecutionCase to the CSV file
+            writeCsvRows(reactivatedCaseList, fw);
+
+        } catch (IOException exception) {
+            log.error("Error creating CSV file - {}", exception.getMessage());
+        }
+        return targetFile;
+    }
+
+    private void writeCsvRows(List<ReactivatedProsecutionCase> reactivatedCaseList, FileWriter fw){
+        reactivatedCaseList.stream()
+                .map(reactivatedProsecutionCase -> reactivatedProsecutionCase.getMaatId() + ","
+                        + reactivatedProsecutionCase.getCaseUrn() + ","
+                        + reactivatedProsecutionCase.getHearingId() + ","
+                        + reactivatedProsecutionCase.getPreviousOutcome() + ","
+                        + reactivatedProsecutionCase.getPreviousOutcomeDate() + ","
+                        + reactivatedProsecutionCase.getDateOfStatusChange() + System.lineSeparator())
+                .forEach(reactivatedCase -> {
+                    try {
+                        fw.append(reactivatedCase);
+                    } catch (IOException ioException) {
+                        log.error("Error writing reactivated case to file  - {}", ioException.getMessage());
+                    }
+                });
+    }
+
+    private File createCsvFile(String fileName) throws IOException {
+        FileAttribute<Set<PosixFilePermission>> attr = PosixFilePermissions.asFileAttribute(PosixFilePermissions.fromString(FILE_PERMISSIONS));
```

Comment:
It's best not to write files to disk when we're just using them to send in emails.

Also, are you cleaning up these files? Theoretically they could potentially bring down the Kubenetes container. By default the temp files here will be created in the container's /tmp directory however K8 containers may need to be explicitly told what to do with these files (even though it should be obvious that they're ephemeral I'm not sure it's guaranteed they are).

Reply:
Snap! @lkm0287 just read your comment, I agree.

> For sending the emails we only need a byte[] representation of the file, so if the anticipated size of the email is small (comfortably sits in the JVM memory) it would best to create the 'file' in memory and send it without the added complexity of writing and deleting files on disk.

Reply:
we are going to send a report to the business once a day, so it will create a file per day and will be deleted after sending. we are good with this.